runtime/pkg/duckdbreplicator/singledb.go

```diff
@@ -0,0 +1,318 @@
+package duckdbreplicator
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/XSAM/otelsql"
+	"github.com/jmoiron/sqlx"
+	"github.com/marcboeker/go-duckdb"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+type singledb struct {
```

Comment:
To keep things simple and ensure we hit similar code paths in as many environments as possible, can this be replaced with a noop or local file bucket for the normal `db` implementation instead?